libs/full/collectives/include/hpx/collectives/all_reduce.hpp

```diff
@@ -219,6 +219,87 @@ namespace hpx { namespace collectives {
     }
 }}    // namespace hpx::collectives
 
+////////////////////////////////////////////////////////////////////////////////
+// compatibility functions
+namespace hpx { namespace lcos {
+
+    template <typename T, typename F>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_reduce is deprecated, use hpx::collectives::all_reduce "
+        "instead")
+    hpx::future<std::decay_t<T>> all_reduce(char const* basename,
+        T&& local_result, F&& op, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::all_reduce(basename,
+            std::forward<T>(local_result), std::forward<F>(op),
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T, typename F>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_reduce is deprecated, use hpx::collectives::all_reduce "
+        "instead")
+    hpx::future<T> all_reduce(char const* basename,
+        hpx::future<T>&& local_result, F&& op,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::all_reduce(basename, f.get(), std::move(op),
```

Comment:
```suggestion
            return hpx::collectives::all_reduce(basename, f.get(), std::forward<F>(op),
```
and others, as pointed out by clang-tidy.

Reply:
I think, the `move` is correct (op was captured by value). The lambda has to be `mutable`, though.